packages/server-core/src/media/storageprovider/ipfs.storage.ts

```diff
@@ -0,0 +1,399 @@
+import * as k8s from '@kubernetes/client-node'
+import { create, IPFSHTTPClient } from 'ipfs-http-client'
+import * as net from 'net'
+import path from 'path'
+import * as stream from 'stream'
+import { concat as uint8ArrayConcat } from 'uint8arrays/concat'
+
+import { FileContentType } from '@xrengine/common/src/interfaces/FileContentType'
+
+import config from '../../appconfig'
+import {
+  BlobStore,
+  PutObjectParams,
+  StorageListObjectInterface,
+  StorageObjectInterface,
+  StorageProviderInterface
+} from './storageprovider.interface'
+
+export class IPFSStorage implements StorageProviderInterface {
+  cacheDomain: string
+
+  _client: IPFSHTTPClient
+  _blobStore: IPFSBlobStore
+  _pathPrefix: string = '/'
+  _apiDomain: string
+
+  doesExist(fileName: string, directoryPath: string): Promise<boolean> {
+    const filePath = path.join(this._pathPrefix, directoryPath, fileName)
+    return this._client.files
+      .stat(filePath)
+      .then(() => true)
+      .catch(() => false)
+  }
+  isDirectory(fileName: string, directoryPath: string): Promise<boolean> {
+    const filePath = path.join(this._pathPrefix, directoryPath, fileName)
+    return this._client.files
+      .stat(filePath)
+      .then((res) => res.type === 'directory')
+      .catch(() => false)
+  }
+  async getObject(key: string): Promise<StorageObjectInterface> {
+    const filePath = path.join(this._pathPrefix, key)
+    const chunks: Uint8Array[] = []
+
+    for await (const chunk of this._client.files.read(filePath)) {
+      chunks.push(chunk)
+    }
+
+    const chunksArray = uint8ArrayConcat(chunks)
+
+    // const decodedData = new TextDecoder().decode(chunksArray).toString();
+    // console.log(decodedData)
+
+    // const file=`https://ipfs.io/ipfs/${hash}`;
+    // const req = await fetch(file, {method:'HEAD'})
+    // console.log(req.headers.get('content-type'))
+
+    return {
+      Body: Buffer.from(chunksArray),
```

Comment:
@HexaField @speigg - our getObject() interface is a full on buffer and not a stream? This seems potentially disastrous, given large enough files (that may overflow node memory constraints). 
But that's unrelated to this PR; I'd love to revisit that later.